hudi-flink/src/main/java/org/apache/hudi/sink/compact/HoodieFlinkCompactor.java

```diff
@@ -112,7 +112,8 @@ public static void main(String[] args) throws Exception {
     }
 
     // get compactionParallelism.
-    int compactionParallelism = Math.min(conf.getInteger(FlinkOptions.COMPACTION_TASKS), compactionPlan.getOperations().size());
+    int compactionParallelism = conf.getInteger(FlinkOptions.COMPACTION_TASKS) == -1
```

Comment:
what will happen if `compactionTasks` set to 0 ? any pre-check needed?

Reply:
It’s fine because Flink don’t allow negative parallelism.